R/apa_print_anova.R

```diff
@@ -61,66 +73,180 @@ apa_print.summary.aov <- function(x, ...) {
+    , p.value = "$p$"
+    , df.residual = "$\\mathit{df}_{\\mathrm{res}}$"
+  )
+
+  canonical_table <- add_custom_effect_sizes(
+    canonical_table = canonical_table
+    , .x = .x
+    , estimate = estimate
+    , mse = mse
+    , observed = observed
+    , intercept = intercept
+  )
+
+  if(!intercept) canonical_table <- canonical_table[canonical_table$term != "(Intercept)", , drop = FALSE]
+
 
-  print_anova(variance_table, ...)
+  ellipsis <- defaults(
+    ellipsis
+    , set.if.null = list(
+      digits = 3L
+      , gt1 = FALSE
+    )
+  )
+  ellipsis$x <- canonical_table
+  beautiful_table <- do.call("beautify", ellipsis)
+
+  glue_apa_results(
+    beautiful_table
+    , est_glue = construct_glue(beautiful_table, "estimate")
+    , stat_glue = construct_glue(beautiful_table, "statistic")
+    , term_names = sanitize_terms(unlabel(canonical_table$term))
+    , in_paren = in_paren
+    , est_first = FALSE
+    , simplify = FALSE
+  )
 }
 
 
 #' @rdname apa_print.aov
 #' @method apa_print Anova.mlm
 #' @export
 
-apa_print.Anova.mlm <- function(x, correction = getOption("papaja.sphericity_correction"), intercept = FALSE, ...) {
+apa_print.Anova.mlm <- function(
+  x
+  , correction = getOption("papaja.sphericity_correction")
+  , intercept = FALSE
+  , ...
+) {
+
   if(correction != "none") {
     summary_x <- summary(x, multivariate = FALSE) # car:::summary.Anova.mlm
   } else { # Corrections are always calculated and can throw warnings; hope I don't regret this
     summary_x <- suppressWarnings(summary(x, multivariate = FALSE)) # car:::summary.Anova.mlm
   }
 
-  apa_print(summary_x, correction = correction, intercept = intercept, ...)
+  apa_print(summary_x, correction = correction, intercept = intercept, .x = x, ...)
 }
 
 
 #' @rdname apa_print.aov
 #' @method apa_print summary.Anova.mlm
 #' @export
 
-apa_print.summary.Anova.mlm <- function(x, correction = getOption("papaja.sphericity_correction"), intercept = FALSE, ...) {
-  variance_table <- arrange_anova(x, correction)
+apa_print.summary.Anova.mlm <- function(
+  x
+  , correction = getOption("papaja.sphericity_correction")
+  , intercept = FALSE
+  , estimate = getOption("papaja.estimate_anova", "ges")
+  , mse = getOption("papaja.mse")
+  , observed = NULL
+  , in_paren = FALSE
+  , ...
+) {
+
+  intercept <- isTRUE(intercept)
+  in_paren  <- isTRUE(in_paren)
+
+  arranged_table <- arrange_anova(x, correction) # arrange_anova.summary.Anova.mlm
+
+  ellipsis <- list(...)
+  .x <- ellipsis$.x
+  ellipsis$.x <- NULL
 
-  print_anova(variance_table, intercept = intercept, ...)
+  canonical_table <- canonize(arranged_table)
+  canonical_table <- add_custom_effect_sizes(
+    canonical_table
+    , estimate = estimate
+    , mse = mse
+    , observed = observed
+    , intercept = intercept
+    , .x = .x
+  )
+
+  # Remove intercept if the user doesn't want it:
+  if(!intercept) canonical_table <- canonical_table[canonical_table$term != "(Intercept)", , drop = FALSE]
```

Comment:
Doesn't this already happen in `add_custom_effect_size()`?

Reply:
Tbh, dunno. There were so many redundancies in the ANOVA code that I thought it might be too far-fetched to remove all of them. Think this PR is a step into the right direction, but more code refactoring will be necessary in the future. But please, let's not mix up this new feature with a refactoring of the ANOVA code, because then it will be implemented in 12 years

Reply:
Ok, now I'm hooked on getting better code: I think I should remove the code from `add_custom_effect_size()` because it really does not really belong there.

Reply:
I also played around with this one. Basically, removing these lines of code leads to side effects with our internal effect-size function. We should open a new issue that aims at really refactoring this function (i.e., making an effect-size function such as `effectsize::eta_squared()`, then we will be able to substantially simplify the ANOVA code.

Reply:
Agreed. Given that you have a better grasp on what's required here, could you open an issue for this?

Reply:
done